atoma-proxy/src/server/handlers/mod.rs

```diff
@@ -82,6 +82,53 @@ pub fn update_state_manager(
     Ok(())
 }
 
+/// Updates the state manager with amount usage for fiat requests.
+///
+/// Updates the amount count for the user with both estimated and actual usage
+///
+/// # Arguments
+///
+/// * `state` - Reference to the application state containing the state manager sender
+/// * `user_id` - Unique identifier of the user
+/// * `estimated_amount` - The estimated amount before processing
+/// * `amount` - The actual amount used
+///
+/// # Returns
+///
+/// Returns `Ok(())` if both updates succeed, or a `AtomaProxyError::InternalError` if either update fails.
+///
+/// # Errors
+///
+/// This function will return an error if:
+/// - The state manager channel is closed
+/// - Either update operation fails to complete
+#[instrument(
+    level = "info",
+    skip_all,
+    fields(user_id, estimated_amount, amount, endpoint)
+)]
+pub fn update_state_manager_fiat(
+    state_manager_sender: &Sender<AtomaAtomaStateManagerEvent>,
+    user_id: i64,
+    estimated_amount: i64,
+    amount: i64,
+    endpoint: &str,
+) -> Result<()> {
+    // Update stack num tokens
+    state_manager_sender
```

Comment:
You need to receive the response back, to know if you actually updated any row, or not. In the latter case, some double spending was tried and this should be properly handled.

Reply:
This function is used only to do refunds, when the computation is done or cancelled. It's not used for locking funds.